awx_collection/tools/templates/tower_module.j2

```diff
@@ -41,8 +41,13 @@ options:
 {% endif %}
       required: {{ item['json']['actions']['POST'][option]['required'] }}
       type: {{ type_map[ item['json']['actions']['POST'][option]['type'] ] }}
-{% if item['json']['actions']['POST'][option].get('default', '') != '' %}
-      default: {{ item['json']['actions']['POST'][option]['default'] }}
+{% if 'default' in item['json']['actions']['POST'][option] %}
+{# for tower_job_template/extra vars, its type is dict but its default is '', so we want to make that {} #}
+{%   if item['json']['actions']['POST'][option]['default'] == '' and type_map[ item['json']['actions']['POST'][option]['type'] ] == 'dict' %}
+      default: {}
```

Comment:
Instead of all this, can we just do `if item['json']['actions']['POST'][option].get('default')`.

that way, `""` `{}`, and anything like that will be handled the same. We don't generally need defaults. I strain to think of a field that should have a default other than `state`.

Reply:
If we stop pulling defaults then we can drop this. In this particular case we have a JSON data structure with a default of '' (extra_vars in a job template). I was getting errors because '' is not a valid JSON structure from an ansible perspective. So I am catching (from the api) a JSON field with a default of '' and making it a default of {} to satiate Ansible. 

Again, I am personally a proponent of trying to make the modules reflect the API with defaults and constraints. My reasoning is that I may be an "ansible" guy and want to know what Tower is  going to do with values I don't set. Sure, I could go look at the Tower API but if I can find out from the module documentation all the better. However, as previously stated, if we were to just put this up in the doc section I would be fine with not enforcing it at the parameter level. It is more work to maintain that but if generate works properly it should be a mute point.